File: Backend/MptRoomAPI/MptRoomAPI/DTO/LessonDTO.cs
using MptRoomAPI.Models.Enums;

namespace MptRoomAPI.DTO
{
    public class LessonDTO
    {
        public int? LessonId { get; set; }
        public int SubjectId { get; set; }
        public int GroupId { get; set; }
        public int TeacherId { get; set; }
        public DayOfWeek DayOfWeek { get; set; }
        public WeekTypeEnum WeekType { get; set; }
        public int LessonNumberId { get; set; }
        public int HousingId { get; set; }
    }
}

File: Frontend/MptRoom/src/app/models/DTO/lesson.dto.ts
import { DayOfWeekEnum } from '../enums/day-of-week.enum';
import { WeekTypeEnum } from '../enums/week-type.enum';

export interface LessonDTO {
  LessonId?: number;
  SubjectId: number;
  GroupId: number;
  TeacherId: number;
  DayOfWeek: DayOfWeekEnum;
  WeekType: WeekTypeEnum;
  LessonNumberId: number;
  HousingId: number;
}
